### PR TITLE
Allow nominating per gamemode on multi-mode/hybrid beatmapsets

### LINKED ISSUE #6689: Multi-mode beatmapsets only require 2 nominations
Currently, a set that has difficulties from all 4 game modes only requires the standard two nominations to proceed into qualified. Logically speaking, shouldn't a hybrid set require 2 nominations per game mode present? If 4 game modes are present, 8 nominations should be required, with enforcement from NAT end to ensure that appropriately qualified BNs are awarding those nominations.

I've asked around as to why this isn't already the case and have been met with a mixture of "the current system doesn't support it" and "it's tradition that it's always been this way". We should address this. The fact that giant hybrid sets are entering Ranked with presumably cover to have less than normal oversight is a huge problem.

### LINKED ISSUE #6875: Outdated hybrid nomination checks
After the change to require 2 BNs for each mode in a mapset, it should now be possible for at most 1 probation BN and at least 1 full BN to nominate their mode. However it seems the restrictions on the site still assume all must be full.

Context: On https://osu.ppy.sh/beatmapsets/1154158, a probationary osu! nominator ([Een Persoon](https://osu.ppy.sh/users/10959501)) received this, while a full osu!taiko nominator ([realy0_](https://osu.ppy.sh/users/8863359)) had nominated:

Could temporarily lift the web restrictions on this, and then update these as part of / following https://github.com/ppy/osu-web/pull/6832.

### PR DESCRIPTION
**Pull request by nekodex**
Following on from #6747, this PR allows BN/NAT members to nominate beatmaps per-gamemode (restricted to the modes the member has been assigned)

Screen Shot 2020-10-30 at 4 54 00 pm

The nomination process for non-hybrid beatmaps remains the same and any hybrid beatmaps that already have a nomination on them will use the old requirements (unless the nomination process is restarted, e.g. via disqualification)

resolves #6689, resolves #6875

### CODE REVIEW

**Comment by nanaya** [comment]
Conflicts :thinking:

**Comment by nanaya** [comment]
I meant something more like this where in non-legacy mode, the only difference is in number of nomination requirement :thinking:

Everything else (playmodes parameter, `xNominationCount` format, etc) stay the same regardless whether or not it's hybrid.

https://gist.github.com/nanaya/6205caba5c24fabfd5df51bf5466f8b0

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/assets/lib/beatmap-discussions/nominator.tsx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        <div className={`${this.bn}`}>
+          <div className={`${this.bn}__header`}>{osu.trans('beatmapsets.nominate.dialog.header')}</div>
+          {content}
+          <div className={`${this.bn}__buttons`}>
+            <BigButton
+              text={osu.trans('beatmaps.nominations.nominate')}
+              icon='fas fa-thumbs-up'
+              isBusy={this.state.loading}
+              props={{
+                disabled: (this.hybridMode() && this.state.selectedModes.length < 1) || this.state.loading,
```
[the remark this answers is not in the document]
Comment (nekodex) [reply resources/assets/lib/beatmap-discussions/nominator.tsx:202]: i don't think so? this is making sure that the user has selected at least one mode to nominate for in the popup dialog before enabling the submit button

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Models/User.php [hunk 2]
```diff
@@ -1446,6 +1460,41 @@ public function visibleGroups()
...
+                    $playmodes = $this->findUserGroup(app('groups')->byIdentifier('bng'), true)->playmodes ?? [];
+                    foreach ($playmodes as $playmode) {
+                        $modes[$playmode] = 'full';
+                    }
+                }
+
+                if ($this->isLimitedBN()) {
+                    $playmodes = $this->findUserGroup(app('groups')->byIdentifier('bng_limited'), true)->playmodes ?? [];
+                    foreach ($playmodes as $playmode) {
+                        // prevent 'full' permission from being overwritten - i.e. if a user is accidentally added to both bng/nat and bng_limited for the same playmode
```
[the remark this answers is not in the document]
Comment (nekodex) [reply app/Models/User.php:1488]: I'll fix this, but apparently NAT and BNG are mutually exclusive anyway 🤷‍♂️

**Review by nanaya — APPROVED** [review]

**Pull request merged**